[qt5-base & qtbase] Fix error on arm64-windows

pedrolcl opened the pull request:
Patch by Oliver Wolff:
https://github.com/qt/qtbase/commit/b66ecb295ecc3981b1c21f01d62c9066f5d6e6ca

qmake has been fixed in 6.8, but all other versions need to be patched for arm64.

The original report is about qt5-base: Build error on arm64-windows. but this PR also adds the same patch to Qt6.

Fixes #37210

pedrolcl committed:
Fix for #37210

Patch by Oliver Wolff:
https://github.com/qt/qtbase/commit/b66ecb295ecc3981b1c21f01d62c9066f5d6e6ca

qmake has been fixed in 6.8, but all other versions need to be patched for arm64.

The original report is about qt5-base: Build error on arm64-windows. but this PR also adds the same patch to Qt6.

cenit commented:
you need to bump (add a 1 to the number there or create it with value 1 if missing) "port-version" for both qt5-base/vcpkg.json and qtbase/vcpkg.json

cenit commented:
and vcpkg team should move to test arm64_windows on real arm64_windows ;)

MonicaLiu0311 commented on ports/qt5-base/patches/qmake-arm64.patch:
```diff
@@ -0,0 +1,31 @@
+     case PROCESSOR_ARCHITECTURE_INTEL:
+         archStr = ProString("x86");
+-- 
+2.16.3
```
```suggestion
2.16.3

```

Add newlines or blank lines as this is a formatting requirement.

MonicaLiu0311 commented on ports/qtbase/qmake-arm64.patch:
```diff
@@ -0,0 +1,31 @@
+     case PROCESSOR_ARCHITECTURE_INTEL:
+         archStr = ProString("x86");
+-- 
+2.16.3
```
```suggestion
2.16.3

```

MonicaLiu0311 commented:
> you need to bump (add a 1 to the number there or create it with value 1 if missing) "port-version" for both qt5-base/vcpkg.json and qtbase/vcpkg.json

1. The above modifications
2. Please commit after modification
3. Run `./vcpkg x-add-version testPort` to generate a series of version according to your commit info (Add `--overwrite-version` when generating other than the first time)
     ```
     vcpkg/versions/
         t-/
             testport.josn (automatically generated)
         baseline.json (automatically add records)
     ```
4. Commit verison information again
5. Push together

pedrolcl committed:
formatting

pedrolcl committed:
port versioning

pedrolcl committed:
bureaucratic requirements

dg0yt commented:
> > and vcpkg team should move to test arm64_windows on real arm64_windows ;)
> 
> Hi! Thanks everyone for the feedback. What do you mean, that the CI build of 'arm64_windows` is not performed on real ARM64 hardware, or in Windows OS?

vcpkg CI does arm64-windows as a cross build on x64-windows. So there is no native testing on arm64 windows hosts. 

(And some ports are hard to adapt to vcpkg's cross-build pattern. `qt5-base` is such a port.)

pedrolcl commented:
Can somebody tell me why the `arm64_osx` build has failed? or is there a build log that I can access?

cenit commented:
> Can somebody tell me why the `arm64_osx` build has failed? or is there a build log that I can access?

in the azure pipeline you can find everything. 
Here the artifacts with the published logs
https://dev.azure.com/vcpkg/c1ee48cb-0df2-4ab3-8384-b1df5a79fe53/_apis/build/builds/108037/artifacts?artifactName=failure%20logs%20for%20arm64-osx&api-version=7.1&%24format=zip

pedrolcl commented:
> > Can somebody tell me why the `arm64_osx` build has failed? or is there a build log that I can access?
> 
> in the azure pipeline you can find everything. Here the artifacts with the published logs https://dev.azure.com/vcpkg/c1ee48cb-0df2-4ab3-8384-b1df5a79fe53/_apis/build/builds/108037/artifacts?artifactName=failure%20logs%20for%20arm64-osx&api-version=7.1&%24format=zip

Thanks. I see that there is a missing header that couldn't be installed. It does not explain why yesterday the same build succeeded.

MonicaLiu0311 commented on ports/qt5-base/patches/qmake-arm64.patch:
```diff
@@ -0,0 +1,31 @@
+From b66ecb295ecc3981b1c21f01d62c9066f5d6e6ca Mon Sep 17 00:00:00 2001
+From: Oliver Wolff <oliver.wolff@qt.io>
+Date: Mon, 15 Jul 2024 09:31:30 +0200
+Subject: [PATCH 1/1] qmake: Add support for arm64 hosts on Windows
+
+Pick-to: 6.8
+Change-Id: I0096300fce870ef4411b071f074588a6420188e0
+Reviewed-by:  Alexey Edelev <alexey.edelev@qt.io>
+Reviewed-by: Joerg Bornemann <joerg.bornemann@qt.io>
+---
+ qmake/library/qmakeevaluator.cpp | 5 +++++
+ 1 file changed, 5 insertions(+)
+
```
Remove comments from the patch to make it minimal.

dg0yt replied:
@cenit The discussion about the comments at the top of patches completed several times with the conclusion to omit them. There is no need to restart that. The accepted format is the output of `git diff`. It is easy to produce when applying changes locally, you don't even have to clone the original git repo. You also get it from the Github web interface with `<PR ULR>.diff` or `<Commut URL>.diff`.

(My question in your opencv4 PR was about even editing that output. AFAICT there is not tool and no `git diff` option to produce it in an automated way. If you want to discuss that topic, better don't do that in other contributor's PRs.)

MonicaLiu0311 commented on ports/qtbase/qmake-arm64.patch:
```diff
@@ -0,0 +1,31 @@
+From b66ecb295ecc3981b1c21f01d62c9066f5d6e6ca Mon Sep 17 00:00:00 2001
+From: Oliver Wolff <oliver.wolff@qt.io>
+Date: Mon, 15 Jul 2024 09:31:30 +0200
+Subject: [PATCH 1/1] qmake: Add support for arm64 hosts on Windows
+
+Pick-to: 6.8
+Change-Id: I0096300fce870ef4411b071f074588a6420188e0
+Reviewed-by:  Alexey Edelev <alexey.edelev@qt.io>
+Reviewed-by: Joerg Bornemann <joerg.bornemann@qt.io>
+---
+ qmake/library/qmakeevaluator.cpp | 5 +++++
+ 1 file changed, 5 insertions(+)
+
```
```suggestion
```

pedrolcl committed:
Apply suggestions from code review

The patches here were not authored by me. These were the original headers:

From b66ecb295ecc3981b1c21f01d62c9066f5d6e6ca Mon Sep 17 00:00:00 2001
From: Oliver Wolff <oliver.wolff@qt.io>
Date: Mon, 15 Jul 2024 09:31:30 +0200
Subject: [PATCH 1/1] qmake: Add support for arm64 hosts on Windows

Co-authored-by: MonicaLiu <110024546+MonicaLiu0311@users.noreply.github.com>

BillyONeal commented:
/azp run

BillyONeal commented:
Rerunning due to network mistake last night.

pedrolcl committed:
port version again

pedrolcl committed:
more bureaucracy

MonicaLiu0311 commented on ports/qt5-base/vcpkg.json:
```diff
@@ -1,6 +1,7 @@
 {
   "name": "qt5-base",
   "version": "5.15.15",
+  "port-version": 2,
```
```suggestion
  "port-version": 1,
```

MonicaLiu0311 commented on ports/qtbase/vcpkg.json:
```diff
@@ -1,6 +1,7 @@
 {
   "name": "qtbase",
   "version": "6.7.3",
+  "port-version": 2,
```
```suggestion
  "port-version": 1,
```

MonicaLiu0311 commented on versions/q-/qt5-base.json:
```diff
@@ -1,5 +1,15 @@
 {
   "versions": [
+    {
+      "git-tree": "a9d66e035a2197a53b7673be7f3a3d5dd00a85bc",
+      "version": "5.15.15",
+      "port-version": 2
+    },
```
```suggestion
```

MonicaLiu0311 replied:
- [ ] Only one version is added to each modified port's versions file.

pedrolcl committed:
revert version bump

pedrolcl committed:
the neverending bureaucracy

pedrolcl commented on versions/q-/qtbase.json:
```diff
@@ -1,5 +1,15 @@
 {
   "versions": [
+    {
+      "git-tree": "fb5e352877c57d32ed6b9afe2dc6584e3d2ed928",
+      "version": "6.7.3",
+      "port-version": 2
+    },
```
```suggestion
```

pedrolcl committed:
Apply suggestions from code review

Co-authored-by: MonicaLiu <110024546+MonicaLiu0311@users.noreply.github.com>
